### PR TITLE
Analyzer: Added CodeFixProvider tests and fixed spacing issue

### PR DESCRIPTION
**Pull request by amdeel**

### CODE REVIEW

File: test/WebJobs.Extensions.DurableTask.Analyzers.Test/Entity/StaticFunctionAnalyzerTests.cs [hunk 1]
```diff
@@ -16,56 +15,49 @@ public class StaticFunctionAnalyzerTests : CodeFixVerifier
...
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
 using Microsoft.Extensions.Logging;
 
 namespace ExternalInteraction
 {
     public static class HireEmployee
     {
         [FunctionName(""HireEmployee"")]
-        public static async Task<Application> RunOrchestrator(
+        public static void RunOrchestrator(
```
Comment (cgillum) [inline test/WebJobs.Extensions.DurableTask.Analyzers.Test/Entity/StaticFunctionAnalyzerTests.cs]: It's a little odd to have an entity function with a method named `RunOrchestrator`. :)

File: test/WebJobs.Extensions.DurableTask.Analyzers.Test/Entity/StaticFunctionAnalyzerTests.cs [hunk 2]
```diff
@@ -16,56 +15,49 @@ public class StaticFunctionAnalyzerTests : CodeFixVerifier
...
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
 using Microsoft.Extensions.Logging;
 
 namespace ExternalInteraction
 {
     public static class HireEmployee
     {
         [FunctionName(""HireEmployee"")]
-        public async Task<Application> RunOrchestrator(
+        public void RunOrchestrator(
```
Comment (cgillum) [inline test/WebJobs.Extensions.DurableTask.Analyzers.Test/Entity/StaticFunctionAnalyzerTests.cs]: It's only wrong to have a non-static entity trigger method when using Entity classes. You can tell that entity classes are used if they call `context.DispatchAsync<T>` where `T` is the class that defines the entity method.

File: test/WebJobs.Extensions.DurableTask.Analyzers.Test/Orchestrator/TimerAnalyzerTests.cs [hunk 3]
```diff
@@ -269,6 +301,390 @@ public static string DirectCall()
...
+
+    namespace VSSample
+    {
+        public static class HelloSequence
+        {
+            [FunctionName('E1_HelloSequence')]
+            public static async Task<List<string>> Run(
+            [OrchestrationTrigger] IDurableOrchestrationContext context)
+            {
+            await context.CreateTimer(context.CurrentUtcDateTime.Add(new TimeSpan(100)), new CancellationToken());
```
Comment (cgillum) [inline test/WebJobs.Extensions.DurableTask.Analyzers.Test/Orchestrator/TimerAnalyzerTests.cs]: Shouldn't there be 4 more spaces before the `await` symbol to match the original source code?

File: src/WebJobs.Extensions.DurableTask.Analyzers/DurableFunctionsCodeFixProvider.cs [hunk 4]
```diff
@@ -12,17 +12,14 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Analyzers
 {
     public abstract class DurableFunctionsCodeFixProvider : CodeFixProvider
     {
-        protected async Task<Document> ReplaceWithIdentifierAsync(Document document, SyntaxNode identifierNode, CancellationToken cancellationToken, string expression)
+        protected async Task<Document> ReplaceWithIdentifierAsync(Document document, SyntaxNode identifierNode, CancellationToken cancellationToken, string identifierString)
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask.Analyzers/DurableFunctionsCodeFixProvider.cs]: I'm not sure I understand what we gain from this being in an abstract class, as opposed to just being a static method. Do we use polymorphism to change behavior, or does it use any `DurableFunctionsCodeFixProvider`-specific state? From what I can tell it does not.

File: src/WebJobs.Extensions.DurableTask.Analyzers/CodefixProviders/Orchestrator/TimerCodeFixProvider.cs [hunk 5]
```diff
@@ -55,32 +56,45 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
 
                     context.RegisterCodeFix(
-                    CodeAction.Create(FixTimerInOrchestrator.ToString(), c => ReplaceWithIdentifierAsync(context.Document, expression, c, newExpression), nameof(TimerCodeFixProvider)),
+                    CodeAction.Create(FixTimerInOrchestrator.ToString(), c => ReplaceWithAwaitExpressionAsync(context.Document, expression, c, newExpression), nameof(TimerCodeFixProvider)),
                     diagnostic);
                 }
             }
         }
 
+        protected async Task<Document> ReplaceWithAwaitExpressionAsync(Document document, SyntaxNode expression, CancellationToken cancellationToken, string expressionString)
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask.Analyzers/CodefixProviders/Orchestrator/TimerCodeFixProvider.cs]: Let's make this static.

File: test/WebJobs.Extensions.DurableTask.Analyzers.Test/Binding/ClientAnalyzerTests.cs [hunk 6]
```diff
@@ -61,26 +73,35 @@ public static class HireEmployee
...
-            [DurableClient] IDurableClient durableClient, [DurableClient] IDurableOrchestrationClient durableOrchestrationClient, 
-            [DurableClient] IDurableEntityClient durableEntityClient, ILogger log)
+            [DurableClient] IDurableEntityClient client,
+            ILogger log)
             {
             }
 }";
+
+        [TestMethod]
+        public void DurableClient_V1_NonIssue()
```
Comment (ConnorMcMahon) [inline test/WebJobs.Extensions.DurableTask.Analyzers.Test/Binding/ClientAnalyzerTests.cs]: I'm not sure I understand what is meant  by the  `_NonIssue` suffix. From what I can tell, this test verifies that the `v1Fix` code does not create any warnings, but that is not immediately clear from the test name.

File: src/WebJobs.Extensions.DurableTask.Analyzers/CodefixProviders/Orchestrator/GuidCodeFixProvider.cs [hunk 7]
```diff
@@ -34,17 +37,31 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
                 {
                     context.RegisterCodeFix(
-                    CodeAction.Create(FixGuidInOrchestrator.ToString(), c => ReplaceWithIdentifierAsync(context.Document, expression, c, variableName + ".NewGuid()"), nameof(GuidCodeFixProvider)),
+                    CodeAction.Create(FixGuidInOrchestrator.ToString(), c => ReplaceWithInvocationExpressionAsync(context.Document, expression, c, variableName + ".NewGuid()"), nameof(GuidCodeFixProvider)),
                     diagnostic);
                 }
             }
-}
+        }
+        private async Task<Document> ReplaceWithInvocationExpressionAsync(Document document, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken, string expressionString)
```
[the remark this answers is not in the document]
Comment (amdeel) [reply src/WebJobs.Extensions.DurableTask.Analyzers/CodefixProviders/Orchestrator/GuidCodeFixProvider.cs]: Yikes.

File: src/WebJobs.Extensions.DurableTask.Analyzers/DurableFunctionsCodeFixProvider.cs (hunk 4 above)
Comment (amdeel) [reply src/WebJobs.Extensions.DurableTask.Analyzers/DurableFunctionsCodeFixProvider.cs]: It doesn't have state. Good point, I'll make it a static class.

**Pull request merged**